core-rust/core-api-server/src/core_api/extractors.rs

```diff
@@ -12,18 +14,35 @@ use super::{client_error, ResponseError};
 
 #[derive(Debug)]
 pub(crate) struct Json<T>(pub T);
-pub use axum::Extension; // Re-export Extension so that it can be used easily
+pub use axum::extract::State; // Re-export Extension so that it can be used easily
 
 #[async_trait]
-impl<B, T> FromRequest<B> for Json<T>
+impl<S, T> FromRequestParts<S> for Json<T>
 where
-    axum::Json<T>: FromRequest<B, Rejection = JsonRejection>,
-    B: HttpBody + Send,
+    axum::Json<T>: FromRequestParts<S, Rejection = JsonRejection>,
+    S: Send + Sync,
 {
     type Rejection = ResponseError<()>;
 
-    async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
-        match axum::Json::<T>::from_request(req).await {
+    async fn from_request_parts(parts: &mut Parts, state: &S) -> Result<Self, Self::Rejection> {
+        match axum::Json::<T>::from_request_parts(parts, state).await {
+            Ok(value) => Ok(Self(value.0)),
+            Err(rejection) => Err(client_error(format!("{:?}", rejection))),
```

Comment:
Looks like this is now throwing the error for request size, so we will probably want to handle it properly and turn it into a 403.

Similarly on line 47? Perhaps this logic can be unified into a method?

Reply:
Fixed. Removed the unused `from_request_parts`.